src/main/java/org/reaktivity/nukleus/http/internal/stream/ClientConnectReplyStream.java

```diff
@@ -102,96 +100,42 @@ public String toString()
         this.windowHandler = this::handleWindow;
     }
 
     @Override
     public void accept(int msgTypeId, DirectBuffer buffer, int index, int length)
     {
-        streamState.accept(msgTypeId, buffer, index, length);
+        handleStream(msgTypeId, buffer, index, length);
     }
 
-    private void handleStreamBeforeBegin(
+    private void handleStream(
         int msgTypeId,
         DirectBuffer buffer,
         int index,
         int length)
     {
-        if (msgTypeId == BeginFW.TYPE_ID)
+        switch (msgTypeId)
         {
+        case BeginFW.TYPE_ID:
             final BeginFW begin = this.factory.beginRO.wrap(buffer, index, index + length);
             handleBegin(begin);
-        }
-        else
-        {
-            handleUnexpected(buffer, index, length);
-        }
-    }
-
-    private void handleStreamWhenBuffering(
-        int msgTypeId,
-        DirectBuffer buffer,
-        int index,
-        int length)
-    {
-        switch (msgTypeId)
-        {
-        case DataFW.TYPE_ID:
-            final DataFW data = this.factory.dataRO.wrap(buffer, index, index + length);
-            handleDataWhenBuffering(data);
-            break;
-        case EndFW.TYPE_ID:
-            handleEndWhenBuffering(buffer, index, length);
             break;
-        case AbortFW.TYPE_ID:
-            final AbortFW abort = this.factory.abortRO.wrap(buffer, index, index + length);
-            handleAbort(abort);
-            break;
-        default:
-            handleUnexpected(buffer, index, length);
-            break;
-        }
-    }
-
-    private void handleStreamWhenNotBuffering(
-        int msgTypeId,
-        DirectBuffer buffer,
-        int index,
-        int length)
-    {
-        switch (msgTypeId)
-        {
         case DataFW.TYPE_ID:
             final DataFW data = this.factory.dataRO.wrap(buffer, index, index + length);
-            handleDataWhenNotBuffering(data);
-            break;
-        case EndFW.TYPE_ID:
-            final EndFW end = this.factory.endRO.wrap(buffer, index, index + length);
-            handleEnd(end);
-            break;
-        case AbortFW.TYPE_ID:
-            final AbortFW abort = this.factory.abortRO.wrap(buffer, index, index + length);
-            handleAbort(abort);
+            handleData(data);
             break;
-        default:
-            handleUnexpected(buffer, index, length);
-            break;
-        }
-    }
-
-    private void handleStreamBeforeEnd(
-        int msgTypeId,
-        DirectBuffer buffer,
-        int index,
-        int length)
-    {
-        switch (msgTypeId)
-        {
         case EndFW.TYPE_ID:
             final EndFW end = this.factory.endRO.wrap(buffer, index, index + length);
-            handleEnd(end);
+            if (slotIndex == NO_SLOT)
+            {
+                handleEnd(end);
+            }
+            else
+            {
+                handleEndWhenBuffering(buffer, index, length);
+            }
```

Comment:
Please simplify this to call `handleEnd(end)` unconditionally, then suggest modifying `handleEnd` as follows:
```
    private void handleEnd(
        EndFW end)
    {
        final long streamId = end.streamId();
        assert streamId == connectReplyId;

        if (slotIndex != NO_SLOT && (responseState == BEFORE_HEADERS || responseState == DATA))
        {
            endDeferred = true;
        }
        else
        {
            // existing handleEnd implementation
        }
    }
```
...and remove `handleEndWhenBuffering` as it is no longer needed.